bin/git-ignore-io

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env bash
+
+gitignore_io_url="https://www.gitignore.io/api/"
+
+default_path="$HOME/.gi_list"
+gi_list=`cat $default_path | tr "," "\n"`
+
+update_gi_list() {
+    curl -L -s  "${gitignore_io_url}/list" > ~/.gi_list
+}
+
+print_in_alphabetical_order() {
+    for i in {a..z};
+    do
+        echo "$gi_list" | grep "^$i" | tr "\n" " "
+        echo
+    done
+}
+
+print_in_table_format() {
+    table_width=5
+    echo "$gi_list" | xargs -n $table_width | column -t
+}
+
+search() {
+    for type in $gi_list;
+    do
+        if [[ $type == *$1* ]]
+        then
+            echo $type
+        fi
+    done
+}
+
+print_last_modified_time() {
+    local gi_list_date=`stat -f "%t%Sm" $default_path`
+    echo "Last update time: $gi_list_date"
+}
+
+gi() {
+    curl -L -s $gitignore_io_url/$1
+}
+
+gi_export() {
+    gi $1 > .gitignore
+}
+
+gi_append() {
+    gi $1 >> .gitignore
+}
+
+show_usage() {
+    echo "Usage:"
+    echo "  git ignore-io <types>...                    Show gitignore template"
+    echo "                [-a|--append] <types>...      Append new .gitignore content to .gitignore under the current directory"
+    echo "                [-e|--export] <types>...      Export new .gitignore to the current directory (The old one will be replaced)"
+    echo "                [-l]                          Print available types in alphabetical order    "
+    echo "                [-L]                          Print available types in table format"
```

Comment:
Forget to update here? Or just haven't finished the update?
If you get all things done, please @ some and let us know.